### PR TITLE
Lay out JavascriptTabModalDialog to related web view in Split view mode

### PR DESCRIPTION
**Pull request by sangwoo108**
In split view mode, it's hard to find which web view triggered javascript tab modal dialog, such as alert, confirm, prompt.

This PR lays out the modal dialogs centered to the related web view so that users can easily recognize which web view triggered the dialog.

Resolves https://github.com/brave/brave-browser/issues/39598

https://github.com/brave/brave-core/assets/5474642/4c9f9746-35c1-4afb-91b2-451c9cc63ec5

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/ui/views/widget/widget_delegate.h
Comment (simonhong) [inline chromium_src/ui/views/widget/widget_delegate.h:23]: can you add some comments why this new callback is introduced instead of using `params_.desired_bounds_delegate`?

Comment (sangwoo108) [reply chromium_src/ui/views/widget/widget_delegate.h:27]: Sure. Done.

**Comment by diracdeltas** [comment]
please get security approval from @bcaller before merge (no need for me to review)

**Comment by bcaller** [comment]
Security review: looks fine.

image

**Comment by bcaller** [comment]
Appears to be a bug with vertical tabs, where the alert placing is calculated incorrectly when vertical tabs are used.

Brave Browser Development 2024-07-09 10 15 58

Brave Browser Development 2024-07-09 10 15 32

**Comment by bcaller** [comment]
Works, I tested creating splits, swapping tab positions, unsplitting. All seemed fine.

**Comment by bcaller** [comment]
Probably OK... alert can end up outside border of browser.
Brave Browser Development 2024-07-09 10 45 20

**Comment by bcaller** [comment]
Was just moving around split / unsplit tabs with multiple alerts, clicking wallet button in right sidebar, not sure what I did exactly but it crashed for the second time. Let me know how to find more crash information to send you if it's relevant to this PR.

```
[49273:259:0709/104635.504309:ERROR:brave_side_panel_coordinator.cc(68)] OnTabStripModelChanged de-activated tab's registry should be observed already!
[0709/104635.641665:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104635.642854:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104635.643398:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104635.643640:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104635.665922:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[49273:259:0709/104635.998022:FATAL:brave_wayback_machine_tab_helper.cc(73)] Check failed: wayback_state_changed_callback_.
0   libbase.dylib                       0x00000001039221b4 base::debug::CollectStackTrace(void const**, unsigned long) + 28
1   libbase.dylib                       0x0000000103909a50 base::debug::StackTrace::StackTrace() + 80
2   libbase.dylib                       0x0000000103819798 logging::LogMessage::Flush() + 152
3   libbase.dylib                       0x000000010381968c logging::LogMessage::~LogMessage() + 36
4   libbase.dylib                       0x00000001037f8fb4 logging::(anonymous namespace)::CheckLogMessage::~CheckLogMessage() + 124
5   libbase.dylib                       0x00000001037f9004 logging::(anonymous namespace)::CheckLogMessage::~CheckLogMessage() + 12
6   libbase.dylib                       0x00000001037f8c90 logging::CheckError::~CheckError() + 44
7   libbase.dylib                       0x00000001037f8cf8 logging::CheckError::~CheckError() + 12
8   libchrome_dll.dylib                 0x00000001118fc144 BraveWaybackMachineTabHelper::SetWaybackStateChangedCallback(base::RepeatingCallback<void (WaybackState)>) + 200
9   libchrome_dll.dylib                 0x000000010f0f63a8 WaybackMachineStateManager::OnTabStripModelChanged(TabStripModel*, TabStripModelChange const&, TabStripSelectionChange const&) + 92
10  libchrome_dll.dylib                 0x0000000111013288 TabStripModel::OnChange(TabStripModelChange const&, TabStripSelectionChange const&) + 500
11  libchrome_dll.dylib                 0x0000000111014858 TabStripModel::SetSelection(ui::ListSelectionModel, TabStripModelObserver::ChangeReason, bool) + 824
12  libchrome_dll.dylib                 0x000000011101445c TabStripModel::ActivateTabAt(int, TabStripUserGestureDetails) + 216
13  libchrome_dll.dylib                 0x0000000110f6c574 non-virtual thunk to Browser::ActivateContents(content::WebContents*) + 68
14  libui_views.dylib                   0x000000011d47ddf4 views::View::Focus() + 44
15  libui_views.dylib                   0x000000011d44d518 views::FocusManager::SetFocusedViewWithReason(views::View*, views::FocusManager::FocusChangeReason) + 544
16  libui_views.dylib                   0x000000011d44cff4 views::FocusManager::AdvanceFocus(bool) + 140
17  libui_views.dylib                   0x000000011d44de34 views::FocusManager::AdvanceFocusIfNecessary() + 100
18  libui_views.dylib                   0x000000011d47b2a4 views::View::SetFocusBehavior(views::View::FocusBehavior) + 116
19  libui_views_controls_webview.dylib  0x000000011cc24440 views::WebView::UpdateCrashedOverlayView() + 104
20  libui_views_controls_webview.dylib  0x000000011cc239a8 views::WebView::SetWebContents(content::WebContents*) + 316
21  libchrome_dll.dylib                 0x00000001113294c0 BrowserView::OnTabDetached(content::WebContents*, bool) + 160
22  libchrome_dll.dylib                 0x0000000110f6a3e0 Browser::OnTabStripModelChanged(TabStripModel*, TabStripModelChange const&, TabStripSelectionChange const&) + 156
23  libchrome_dll.dylib                 0x000000010f0525cc BraveBrowser::OnTabStripModelChanged(TabStripModel*, TabStripModelChange const&, TabStripSelectionChange const&) + 32
24  libchrome_dll.dylib                 0x0000000111013288 TabStripModel::OnChange(TabStripModelChange const&, TabStripSelectionChange const&) + 500
25  libchrome_dll.dylib                 0x0000000111013c24 TabStripModel::SendDetachWebContentsNotifications(TabStripModel::DetachNotifications*) + 472
26  libchrome_dll.dylib                 0x0000000111017a9c TabStripModel::CloseTabs(base::span<content::WebContents* const, 18446744073709551615ul, content::WebContents* const*>, unsigned int) + 1376
27  libchrome_dll.dylib                 0x0000000111018054 TabStripModel::CloseWebContentsAt(int, unsigned int) + 188
28  libcontent.dylib                    0x0000000108ebe128 non-virtual thunk to content::WebContentsImpl::Close() + 116
29  libcontent.dylib                    0x000000010832d2b4 base::internal::Invoker<base::internal::FunctorTraits<void (content::background_fetch::(anonymous namespace)::CanCreateRegistrationTask::*&&)(blink::mojom::BackgroundFetchError), base::WeakPtr<content::background_fetch::(anonymous namespace)::CanCreateRegistrationTask>&&, blink::mojom::BackgroundFetchError&&>, base::internal::BindState<true, true, false, void (content::background_fetch::(anonymous namespace)::CanCreateRegistrationTask::*)(blink::mojom::BackgroundFetchError), base::WeakPtr<content::background_fetch::(anonymous namespace)::CanCreateRegistrationTask>, blink::mojom::BackgroundFetchError>, void ()>::RunOnce(base::internal::BindStateBase*) + 92
30  libblink_common.dylib               0x000000011af62398 base::OnceCallback<void ()>::Run() && + 68
31  libblink_common.dylib               0x000000011b13ec08 blink::mojom::LocalMainFrame_ClosePage_ForwardToCallback::Accept(mojo::Message*) + 56
32  libmojo_public_cpp_bindings.dylib   0x00000001032afab4 mojo::InterfaceEndpointClient::HandleValidatedMessage(mojo::Message*) + 1480
33  libmojo_public_cpp_bindings.dylib   0x00000001032b6a70 mojo::MessageDispatcher::Accept(mojo::Message*) + 232
34  libmojo_public_cpp_bindings.dylib   0x00000001032b14b4 mojo::InterfaceEndpointClient::HandleIncomingMessage(mojo::Message*) + 100
35  libipc.dylib                        0x0000000104d14a8c IPC::ChannelAssociatedGroupController::AcceptOnEndpointThread(mojo::Message, IPC::(anonymous namespace)::ScopedUrgentMessageNotification) + 464
36  libipc.dylib                        0x0000000104d15178 base::internal::Invoker<base::internal::FunctorTraits<void (IPC::ChannelAssociatedGroupController::*&&)(mojo::Message, IPC::(anonymous namespace)::ScopedUrgentMessageNotification), IPC::ChannelAssociatedGroupController*&&, mojo::Message&&, IPC::(anonymous namespace)::ScopedUrgentMessageNotification&&>, base::internal::BindState<true, true, false, void (IPC::ChannelAssociatedGroupController::*)(mojo::Message, IPC::(anonymous namespace)::ScopedUrgentMessageNotification), scoped_refptr<IPC::ChannelAssociatedGroupController>, mojo::Message, IPC::(anonymous namespace)::ScopedUrgentMessageNotification>, void ()>::RunOnce(base::internal::BindStateBase*) + 104
37  libbase.dylib                       0x00000001037f61f8 base::OnceCallback<void ()>::Run() && + 68
38  libbase.dylib                       0x0000000103889210 base::TaskAnnotator::RunTaskImpl(base::PendingTask&) + 272
39  libbase.dylib                       0x00000001038b589c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::LazyNow*) + 984
40  libbase.dylib                       0x00000001038b5160 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() + 100
41  libbase.dylib                       0x00000001039341d8 base::MessagePumpCFRunLoopBase::RunWork() + 164
42  libbase.dylib                       0x000000010392c2b4 base::apple::CallWithEHFrame(void () block_pointer) + 16
43  libbase.dylib                       0x0000000103933728 base::MessagePumpCFRunLoopBase::RunWorkSource(void*) + 68
44  CoreFoundation                      0x0000000199df64d8 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
45  CoreFoundation                      0x0000000199df646c __CFRunLoopDoSource0 + 176
46  CoreFoundation                      0x0000000199df61dc __CFRunLoopDoSources0 + 244
47  CoreFoundation                      0x0000000199df4dc8 __CFRunLoopRun + 828
48  CoreFoundation                      0x0000000199df4434 CFRunLoopRunSpecific + 608
49  HIToolbox                           0x00000001a459819c RunCurrentEventLoopInMode + 292
50  HIToolbox                           0x00000001a4597fd8 ReceiveNextEventCommon + 648
51  HIToolbox                           0x00000001a4597d30 _BlockUntilNextEventMatchingListInModeWithFilter + 76
52  AppKit                              0x000000019d653d68 _DPSNextEvent + 660
53  AppKit                              0x000000019de49808 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 700
54  libchrome_dll.dylib                 0x000000010f681100 __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke + 64
55  libbase.dylib                       0x000000010392c2b4 base::apple::CallWithEHFrame(void () block_pointer) + 16
56  libchrome_dll.dylib                 0x000000010f681048 -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 188
57  AppKit                              0x000000019d64709c -[NSApplication run] + 476
58  libbase.dylib                       0x0000000103934ed0 base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 408
59  libbase.dylib                       0x00000001039330a4 base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 140
60  libbase.dylib                       0x00000001038b64c4 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) + 528
61  libbase.dylib                       0x000000010385f6b8 base::RunLoop::Run(base::Location const&) + 364
62  libcontent.dylib                    0x00000001086fb894 content::BrowserMainLoop::RunMainMessageLoop() + 180
63  libcontent.dylib                    0x00000001086fd458 content::BrowserMainRunnerImpl::Run() + 48
64  libcontent.dylib                    0x00000001086f8f60 content::BrowserMain(content::MainFunctionParams) + 152
65  libcontent.dylib                    0x000000010936e918 content::RunBrowserProcessMain(content::MainFunctionParams, content::ContentMainDelegate*) + 280
66  libcontent.dylib                    0x000000010936fef8 content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams, bool) + 1008
67  libcontent.dylib                    0x000000010936f9e0 content::ContentMainRunnerImpl::Run() + 548
68  libcontent.dylib                    0x000000010936df2c content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) + 1204
69  libcontent.dylib                    0x000000010936e180 content::ContentMain(content::ContentMainParams) + 112
70  libchrome_dll.dylib                 0x000000010e1d21d0 ChromeMain + 472
71  Brave Browser Development           0x0000000102a6c7e0 main + 288
72  dyld                                0x000000019998e0e0 start + 2360
Task trace:
0   libipc.dylib                        0x0000000104d0f814 IPC::ChannelAssociatedGroupController::Accept(mojo::Message*) + 880
1   libmojo_public_system_cpp.dylib     0x000000010332cf94 mojo::SimpleWatcher::Context::Notify(unsigned int, MojoHandleSignalsState, unsigned int) + 224
Crash keys:
  "tabdrag-event-source" = "mouse"
  "num-extensions" = "0"
  "switch-11" = "--component-updater=url-source=https://go-updater.brave.com/exte"
  "commandline-enabled-feature-1" = "BraveSplitView"
  "amfi-status" = "rv=0 status=0x0 allow_everything=0"
  "reentry_guard_tls_slot" = "364"
  "switch-10" = "--variations-insecure-server-url=https://variations.brave.com/se"
  "switch-9" = "--variations-server-url=https://variations.brave.com/seed"
  "switch-8" = "--sync-url=https://sync-v2.brave.software/v2"
  "switch-7" = "--lso-url=https://no-thanks.invalid"
  "switch-6" = "--origin-trial-public-key=[[[[["
  "switch-5" = "--enable-distillability-service"
  "switch-4" = "--enable-dom-distiller"
  "switch-3" = "--disable-domain-reliability"
  "switch-2" = "--user-data-dir=/Users/b/brave/test/dev/"
  "switch-1" = "--disable-brave-update"
  "num-switches" = "16"
  "osarch" = "arm64"
  "pid" = "49273"
  "ptype" = "browser"

[0709/104636.145136:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104636.145949:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104636.146438:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104636.146662:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104636.166840:WARNING:process_memory_mac.cc(94)] mach_vm_read(0x102ebc000, 0x8000): (os/kern) invalid address (1)
[0709/104636.495080:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
```

**Review by bcaller — CHANGES_REQUESTED** [review]
From the UI side I don't have any security concerns. The placement bug is not security relevant. The crash does need investigation though. Please let me know how to help you diagnose it.

**Comment by sangwoo108** [comment]
image

Thanks @bcaller . Now it should be fine in vertical tab mode.

**Comment by sangwoo108** [comment]
@simonhong , do you know what would be proper fix for https://github.com/brave/brave-core/pull/24540#issuecomment-2216165817

**Comment by simonhong** [comment]
> @simonhong , do you know what would be proper fix for [#24540 (comment)](https://github.com/brave/brave-core/pull/24540#issuecomment-2216165817) ?

Well.. it looks like that failure happens while tab detaching/attaching.
Regardless of this PR, I could make dcheck failure when attaching and detaching with same tile several times.
IMO, this failure could be investigated more in a separated issue.

**Comment by simonhong** [comment]
modal dialog is rendered over the another application.
Screenshot 2024-07-09 at 3 37 06 PM

**Comment by sangwoo108** [comment]
> modal dialog is rendered over the another application. Screenshot 2024-07-09 at 3 37 06 PM

Thanks, let me try adjusting z-order level

**Comment by sangwoo108** [comment]
> > modal dialog is rendered over the another application. <img alt="Screenshot 2024-07-09 at 
> Thanks, let me try adjusting z-order level

Changed to use SetZOrderSublevel

**Review by bcaller — APPROVED** [review]

**Review by simonhong — APPROVED** [review]
++

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:18]: Please add comments documenting this class.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:45]: is `alerting_web_contents_` guaranteed to out-live this class?

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:32]: Please consider a comment in this header explaining what could trigger a force closed callback.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:38]: This function will never return `nullptr` by the looks of the implementation, therefore it is better to have it returning a reference, i.e. `web_modal::WebContentsModalDialogHost&`.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:18]: It looks a bit odd to me that we would use `ViewViews`. Granted I'm not that familiar with UI terminology, but I'm struggling to see what this repetition is supposed to say.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:25]: If this value cannot be passed as `nullptr`, pass this a reference.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:26]: If this value cannot be `nullptr`, pass this a reference.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:45]: if `alerting_web_contents_` is guaranteed to outlive this class' instance, and is guaranteed to never be `nullptr`, prefer the use of `raw_ref`.

File: chromium_src/chrome/browser/ui/views/javascript_tab_modal_dialog_view_views.cc
Comment (cdesouza-chromium) [inline chromium_src/chrome/browser/ui/views/javascript_tab_modal_dialog_view_views.cc:8]: Maybe it would be good to have a comment here indicating why this substitution is necessary.

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.cc
Comment (cdesouza-chromium) [inline browser/ui/views/brave_javascript_tab_modal_dialog_view_views.cc:109]: is there any reason in particular why this function tests for `!widget`, rather than `CHECK`ing it like the other places?

File: browser/ui/views/frame/brave_browser_view.h
Comment (cdesouza-chromium) [inline browser/ui/views/frame/brave_browser_view.h:134]: This declaration seems to be at odds with the style guide [1] which indicates that `auto` deducted return types shouldn't be used on header files.

> use it only if the function or lambda has a very narrow scope, because functions with deduced return types don't define abstraction boundaries: the implementation is the interface. In particular, public functions in header files should almost never have deduced return types.

[https://google.github.io/styleguide/cppguide.html#Type_deduction]

File: browser/ui/views/frame/brave_browser_view.cc
Comment (cdesouza-chromium) [inline browser/ui/views/frame/brave_browser_view.cc:325]: It feels like `browser_view_` should be set only during construction of `BraveContentsLayoutManager`, and it should be passed as a reference, as `this` is never `nullptr`. I would also use `raw_ref` with `browser_view_`.

File: chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h
Comment (cdesouza-chromium) [inline chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h:18]: maybe we should add a comment about the purpose of these overrides.

**Review by sangwoo108 — COMMENTED** [review]
Thanks @cdesouza-chromium . Could you take another look?

**Comment by sangwoo108** [comment]
I found that using alerting_contents could be a problem. alerting_web_contents might not be in tab model. Because in case of chrome app's <webview>, parent web contents is different from alerting_web_contents. So changed to use parent web contents.

https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.h;l=78;drc=7f04a8e033c23dede6beae129cd212e6d4473d72

File: browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h
Comment (cdesouza-chromium) [reply browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h:35]: if the base class has `dialog_force_closed_callback`, there's no need to document.

File: chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h:20]: I think this comment can be improved a bit. Right now it is describing something I already can see just by looking at the override itself. It would better if we could describe "why" we are adding these methods in particular.

**Review by cdesouza-chromium — APPROVED** [review]
LGTM % changes.

**Review by bcaller — APPROVED** [review]

File: chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/frame/browser_view_layout.h:20]: Good call. Updated.

**Pull request merged**
